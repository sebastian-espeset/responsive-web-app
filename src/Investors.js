import React from 'react';
import { StyledInvestorsContainer, StyledInvestorsHeadline, StyledInvestorsLogoContainer } from './styles/Investors.style';
import drivecapital from './assets/drivecapital.png';
import georgianpartners from './assets/georgianpartners.png';
import kleinerperkins from './assets/kleinerperkins.png';
import lewisandclark from './assets/lewisandclarkventures.png';

export default function Investors() {
  return (
    <StyledInvestorsContainer>
      <StyledInvestorsHeadline data-testid='investors-headline-test-id'>Our Investors</StyledInvestorsHeadline>
      <div>────────</div>
      <StyledInvestorsLogoContainer>
        <img src={drivecapital} alt="drive captial logo"></img>
        <img src={georgianpartners} alt="georgian partners logo"></img>
        <img src={kleinerperkins} alt="kleiner perkins logo"></img>
        <img src={lewisandclark} alt="lewis and clark ventures logo"></img>
      </StyledInvestorsLogoContainer>
    </StyledInvestorsContainer>
  )
}
