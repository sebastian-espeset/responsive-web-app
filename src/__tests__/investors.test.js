import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Investors from '../Investors';

afterEach(()=>{
    cleanup();
});

test('investors section headline should render',()=>{
    //Arrange
    render(<Investors/>)
    //Act
    const investorsHeadline = screen.getByTestId('investors-headline-test-id');
    //Assert
    expect(investorsHeadline).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('investors component to match snapshot',()=>{
    const tree = renderer.create(<Investors/>).toJSON();
    expect(tree).toMatchSnapshot();
})