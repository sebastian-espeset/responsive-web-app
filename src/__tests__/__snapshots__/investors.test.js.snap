// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`investors component to match snapshot 1`] = `
<div
  className="sc-bczRLJ chjraK"
>
  <h1
    className="sc-gsnTZi ekMsiu"
    data-testid="investors-headline-test-id"
  >
    Our Investors
  </h1>
  <div>
    ────────
  </div>
  <div
    className="sc-dkzDqf iiLzdx"
  >
    <img
      alt="drive captial logo"
      src="drivecapital.png"
    />
    <img
      alt="georgian partners logo"
      src="georgianpartners.png"
    />
    <img
      alt="kleiner perkins logo"
      src="kleinerperkins.png"
    />
    <img
      alt="lewis and clark ventures logo"
      src="lewisandclarkventures.png"
    />
  </div>
</div>
`;
